packages/beacon-node/src/chain/seenCache/seenAttesters.ts

```diff
@@ -1,18 +1,24 @@
 import {Epoch, ValidatorIndex} from "@lodestar/types";
 import {MapDef} from "@lodestar/utils";
 
-// The next, current and previous epochs. We require the next epoch due to the
-// `MAXIMUM_GOSSIP_CLOCK_DISPARITY`. We require the previous epoch since the
-// specification delcares:
+// How many *non future* epoch we intend to keep for SeenAttesters.
+// Pre and post deneb specs require us to accept attestations from current and
+// previous epoch.
 //
-// ```
-// aggregate.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE
-//      >= current_slot >= aggregate.data.slot
-// ```
+// Pre-deneb:
+// - `attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= attestation.data.slot`
 //
-// This means that during the current epoch we will always accept an attestation
-// from at least one slot in the previous epoch.
-const MAX_EPOCHS = 3;
+// Post-deneb:
+// - `attestation.data.slot <= current_slot`
+// - `compute_epoch_at_slot(attestation.data.slot) in (get_previous_epoch(state), get_current_epoch(state))`
+//
+// When factored in MAXIMUM_GOSSIP_CLOCK_DISPARITY, it is possible we keep 3 epochs of SeenAttesters:
+// previous, current and future epoch. This constant is solely used to calculate `lowestPermissableEpoch`
+// which prunes anything older than it.
+//
+// assuming we're at epoch 100 while all other nodes at epoch 99, they all accept attestations at epoch 98, 99. If MAX_RETAINED_EPOCH = 2 then our lowestPermissableEpoch is 98 which is fine
+// assuming we're at epoch 99 while all other nodes at epoch 100, they all accept attestations at epoch 99, 100. If MAX_RETAINED_EPOCH = 2 then lowestPermissableEpoch is 97 which is more than enough
```

Comment:
should add a line breaks here, not sure if we can enforce a line width 120 on comments, but prettier doesn't enforce it there

https://github.com/ChainSafe/lodestar/blob/83fd80d5759b4b3f8aa9332ac53af2c7e1977d8a/.prettierrc.js#L2